source/llassetgen/include/llassetgen/Vec2.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+namespace llassetgen {
+    template <class T>
+    struct Vec2 {
+        T x = 0;
+        T y = 0;
+
+        Vec2() = default;
+
+        template <class T2>
```

Comment:
What is the second template for?

Reply:
I mostly kept it because it was defined that way in the original `PositionType`. The only place I know where this is necessary is when moving the arguments (`T2&&`), but that is definitely not necessary if we only plan on using integers and floats with this class.

I'll be removing the second template, we can always add it back if necessary.